tools/generator/Tests/expected/CovariantReturnTypes/CovariantReturnTypes.xml

```diff
@@ -0,0 +1,88 @@
+			<method abstract="true" deprecated="not deprecated" final="false" name="getObject" native="false" return="java.lang.Object" static="false" synchronized="false" visibility="public">
+			</method>
+		</interface>
+		<!--
+			public class CovariantInterfaceImplementation implements CovariantInterface {
+				@Override
+				String getObject();
+			}
+		-->
+		<class abstract="false" deprecated="not deprecated" extends="java.lang.Object" extends-generic-aware="java.lang.Object" final="false" name="CovariantInterfaceImplementation" static="false" visibility="public">
+			<implements name="covariant.returntypes.CovariantInterface" name-generic-aware="covariant.returntypes.CovariantInterface" />
+			<constructor deprecated="not deprecated" final="false" name="CovariantInterfaceImplementation" static="false" visibility="public" />
+			<method abstract="false" deprecated="not deprecated" final="false" name="getObject" native="false" return="java.lang.String" static="false" synchronized="false" visibility="public">
+			</method>
+		</class>
+		<!--
+			public class CovariantClass {
+				Object getObject();
+			}
+		-->
+		<class abstract="false" deprecated="not deprecated" extends="java.lang.Object" extends-generic-aware="java.lang.Object" final="false" name="CovariantClass" static="false" visibility="public">
+			<method abstract="false" deprecated="not deprecated" final="false" name="getObject" native="false" return="java.lang.Object" static="false" synchronized="false" visibility="public">
+			</method>
+		</class>
+		<!--
+			public class CovariantSubclass extends CovariantClass {
+				@Override
+				String getObject();
+			}
+		-->
+		<class abstract="false" deprecated="not deprecated" extends="covariant.returntypes.CovariantClass" extends-generic-aware="covariant.returntypes.CovariantClass" final="false" name="CovariantSubclass" static="false" visibility="public">
+			<constructor deprecated="not deprecated" final="false" name="CovariantInterfaceImplementation" static="false" visibility="public" />
+			<method abstract="false" deprecated="not deprecated" final="false" name="getObject" native="false" return="java.lang.String" static="false" synchronized="false" visibility="public">
+			</method>
+		</class>
+		<!-- 
+			public interface GenericInterface<T> {
+				T getObject();
+			}
+		-->
+		<interface abstract="true" deprecated="not deprecated" final="false" name="GenericInterface" static="false" visibility="public">
+			<typeParameters>
+				<typeParameter name="T" classBound="java.lang.Object" interfaceBounds="" />
+			</typeParameters>
+			<method abstract="true" deprecated="not deprecated" final="false" name="getObject" native="false" return="T" static="false" synchronized="false" visibility="public">
+			</method>
+		</interface>
+		<!--
+			public class GenericImplementation implements GenericInterface<Object> {
```

Comment:
For good measure/coverage/"future proofing", you should also try binding:

```java
public class GenericStringImplementation implements GenericInterface<String> {
    public String getObject() {return "";}
}
```

...just to make sure that does something "reasonable."

Reply:
Well, this example breaks... even without my changes 😞 

It looks like there is not an implementation for explicit interfaces on properties, like we have for methods: https://github.com/xamarin/java.interop/blob/79a8e1ee93673aa2f8ef9f2ea8cb9bebdc1394df/tools/generator/Method.cs#L598

We need a second `foreach` loop here for properties: https://github.com/xamarin/java.interop/blob/461878798f8ec076e2f56c0141aa7b605e911fb0/tools/generator/ClassGen.cs#L474

I think I should redo these covariant tests to just use methods (maybe just have one case for a property). After this gets merged, I can revisit the issues with generic interfaces + properties and make specific test cases for those.